hacep/src/main/java/org/kie/hacep/consumer/CommandHandler.java

```diff
@@ -50,37 +52,53 @@ public CommandHandler(KieSessionContext kieSessionContext,
         this.producer = producer;
     }
 
+    @Override
+    public void visit(FireAllRulesCommand command) {
+        int fires = kieSessionContext.getKieSession().fireAllRules();
+        producer.produceSync(config.getKieSessionInfosTopicName(), command.getId(), fires);
+    }
+
     @Override
     public void visit(InsertCommand command) {
         RemoteFactHandle remoteFH = command.getFactHandle();
         FactHandle fh = kieSessionContext.getKieSession().getEntryPoint(command.getEntryPoint()).insert(remoteFH.getObject());
         kieSessionContext.getFhManager().registerHandle(remoteFH, fh);
-        kieSessionContext.getKieSession().fireAllRules();
+        if (command.isAutoFire()) {
```

Comment:
I think it is wrong to have `isAutoFire` as property of the command. This means that status (fire or wait) is stored on client side and breaks in case of multiple clients.
I think this property should be stored inside drools executor and also inside snapshot status.
Wdyt?